test_rmw_implementation/test/test_wait_set.cpp

```diff
@@ -0,0 +1,322 @@
+  EXPECT_EQ(ret, RMW_RET_TIMEOUT) << rcutils_get_error_string().str;
+  rmw_reset_error();
+
+  // Used a valid wait_set and subscription with no timeout
+  ret = rmw_wait(
+    subscriptions, nullptr, nullptr, nullptr, nullptr, wait_set,
+    &timeout_argument_zero);
+  EXPECT_EQ(ret, RMW_RET_TIMEOUT) << rcutils_get_error_string().str;
+  rmw_reset_error();
+
+  // Used a valid wait_set, subscription and guard_conditions with 100ms timeout
+  ret = rmw_wait(
+    subscriptions, guard_conditions, nullptr, nullptr, nullptr, wait_set,
+    &timeout_argument);
+  EXPECT_EQ(ret, RMW_RET_TIMEOUT) << rcutils_get_error_string().str;
+  rmw_reset_error();
+
+  // Used a valid wait_set, subscription and guard_conditions with no timeout
+  ret = rmw_wait(
+    subscriptions, guard_conditions, nullptr, nullptr, nullptr, wait_set,
+    &timeout_argument_zero);
+  EXPECT_EQ(ret, RMW_RET_TIMEOUT) << rcutils_get_error_string().str;
+  rmw_reset_error();
+
+  // Used a valid wait_set, subscription, guard_conditions and services with 100ms timeout
+  ret = rmw_wait(
+    subscriptions, guard_conditions, services, nullptr, nullptr, wait_set,
+    &timeout_argument);
+  EXPECT_EQ(ret, RMW_RET_TIMEOUT) << rcutils_get_error_string().str;
+  rmw_reset_error();
+
+  // Used a valid wait_set, subscription, guard_conditions and services with no timeout
+  ret = rmw_wait(
+    subscriptions, guard_conditions, services, nullptr, nullptr, wait_set,
+    &timeout_argument_zero);
+  EXPECT_EQ(ret, RMW_RET_TIMEOUT) << rcutils_get_error_string().str;
+  rmw_reset_error();
+
+  // Used a valid wait_set, subscription, guard_conditions, services and clients with 100ms timeout
+  ret = rmw_wait(
+    subscriptions, guard_conditions, services, clients, nullptr, wait_set,
+    &timeout_argument);
+  EXPECT_EQ(ret, RMW_RET_TIMEOUT) << rcutils_get_error_string().str;
+  rmw_reset_error();
+
+  // Used a valid wait_set, subscription, guard_conditions, services and clients with no timeout
+  ret = rmw_wait(
+    subscriptions, guard_conditions, services, clients, nullptr, wait_set,
+    &timeout_argument_zero);
+  EXPECT_EQ(ret, RMW_RET_TIMEOUT) << rcutils_get_error_string().str;
+  rmw_reset_error();
+
+  // Used a valid wait_set, subscription, guard_conditions, services, clients and events with 100ms
+  // timeout
+  ret = rmw_wait(
+    subscriptions, guard_conditions, services, clients, events, wait_set,
+    &timeout_argument);
+  EXPECT_EQ(ret, RMW_RET_TIMEOUT) << rcutils_get_error_string().str;
+  rmw_reset_error();
+
+  // Used a valid wait_set, subscription, guard_conditions, services, clients and events with no
+  // timeout
+  ret = rmw_wait(
+    subscriptions, guard_conditions, services, clients, events, wait_set,
+    &timeout_argument_zero);
+  EXPECT_EQ(ret, RMW_RET_TIMEOUT) << rcutils_get_error_string().str;
+  rmw_reset_error();
+
+  const char * implementation_identifier = wait_set->implementation_identifier;
+  wait_set->implementation_identifier = "rmw_baddss_cpp";
```

Comment:
@ahcorde nit^N: pretty much everywhere else, we've been using "not-an-rmw-implementation-identifier" for consistency.